### PR TITLE
[doc][hackathon] To add AdamW Optimizer to the documentation

### PR DESCRIPTION
**Pull request by iramazanli**
It has been discussed before that adding description of Optimization algorithms to PyTorch Core documentation may result in a nice Optimization research tutorial. In the following tracking issue we mentioned about all the necessary algorithms and links to the originally published paper  https://github.com/pytorch/pytorch/issues/63236.

In this PR we are adding description of AdamW Algorithm to the documentation.  For more details, we refer to the paper  here https://arxiv.org/abs/1711.05101

AdamWalgo

cc @vincentqb @iramazanli

### CODE REVIEW

File: torch/optim/adamw.py [hunk 1]
```diff
@@ -6,8 +6,37 @@
...
+            &\textbf{input}      : \gamma \text{(lr)}, \: \beta_1, \beta_2
+                \text{(betas)}, \: \theta_0 \text{(params)}, \: f(\theta) \text{(objective)},    \\
+            &\hspace{13mm}      \lambda \text{(weight decay)}, \:  amsgrad                       \\
+            &\textbf{initialize} : m_0 \leftarrow 0 \text{ (first moment)}, v_0 \leftarrow 0
+                \text{ ( second moment)}, \: \widehat{v_0}^{max}\leftarrow 0              \\[-1.ex]
+            &\rule{110mm}{0.4pt}                                                                 \\
+            &\textbf{for} \: t=1 \: \textbf{to} \: \ldots \: \textbf{do}                         \\
+            &\hspace{5mm}g_t           \leftarrow   \nabla_{\theta} f_t (\theta_{t-1})           \\
+            &\hspace{5mm}\textbf{if} \: \lambda \neq 0                                           \\
+            &\hspace{10mm} g_t \leftarrow g_t + \lambda \theta_{t-1}                             \\
```
Comment (albanD) [inline torch/optim/adamw.py:21]: Is that correct? We don't actually do weight decay this way for this one no?

File: torch/optim/adamw.py [hunk 2]
```diff
@@ -6,8 +6,37 @@
...
+            &\hspace{5mm}\widehat{m_t} \leftarrow   m_t/\big(1-\beta_1^t \big)                   \\
+            &\hspace{5mm}\widehat{v_t} \leftarrow   v_t/\big(1-\beta_2^t \big)                   \\
+            &\hspace{5mm}\textbf{if} \: amsgrad                                                  \\
+            &\hspace{10mm}\widehat{v_t}^{max} \leftarrow \mathrm{max}(\widehat{v_t}^{max},
+                \widehat{v_t})                                                                   \\
+            &\hspace{10mm}\theta_t \leftarrow \theta_{t-1} - \gamma \widehat{m_t}/
+                \big(\sqrt{\widehat{v_t}^{max}} + \epsilon + \lambda \theta_{t-1}  \big)         \\
+            &\hspace{5mm}\textbf{else}                                                           \\
+            &\hspace{10mm}\theta_t \leftarrow \theta_{t-1} - \gamma \widehat{m_t}/
+                \big(\sqrt{\widehat{v_t}} + \epsilon + \lambda \theta_{t-1}  \big)               \\
```
Comment (albanD) [inline torch/optim/adamw.py:33]: Where is the \lambda \theta_{t-1} coming from?

File: torch/optim/adamw.py (hunk 1 above)
Comment (iramazanli) [reply torch/optim/adamw.py:21]: yes, you're completely right !

File: torch/optim/adamw.py (hunk 2 above)
Comment (iramazanli) [reply torch/optim/adamw.py]: i think it shouldn't exist, you're right again !

File: torch/optim/adamw.py [hunk 3]
```diff
@@ -6,8 +6,37 @@
...
+            &\textbf{input}      : \gamma \text{(lr)}, \: \beta_1, \beta_2
+                \text{(betas)}, \: \theta_0 \text{(params)}, \: f(\theta) \text{(objective)},
+                \: \epsilon \text{ (epsilon)}                                                    \\
+            &\hspace{13mm}      \lambda \text{(weight decay)}, \:  amsgrad                       \\
+            &\textbf{initialize} : m_0 \leftarrow 0 \text{ (first moment)}, v_0 \leftarrow 0
+                \text{ ( second moment)}, \: \widehat{v_0}^{max}\leftarrow 0              \\[-1.ex]
+            &\rule{110mm}{0.4pt}                                                                 \\
+            &\textbf{for} \: t=1 \: \textbf{to} \: \ldots \: \textbf{do}                         \\
+            &\hspace{5mm}g_t           \leftarrow   \nabla_{\theta} f_t (\theta_{t-1})           \\
+            &\hspace{5mm} \theta_t \leftarrow \theta_{t-1} - \lambda \theta_{t-1}                \\
```
Comment (albanD) [inline torch/optim/adamw.py:21]: This is missing a multiplication by `\gamma` no?

Comment (iramazanli) [reply torch/optim/adamw.py]: yes, i added it. thanks for pointing it out

**Review by albanD — APPROVED** [review]
ok
